src/bootstrap/build/util.rs

```diff
@@ -16,6 +16,26 @@ use std::process::Command;
 use bootstrap::{dylib_path, dylib_path_var};
 use filetime::FileTime;
 
+#[cfg(not(windows))]
+pub fn build_path(path: &str) -> PathBuf {
+    PathBuf::from(path)
+}
+
+#[cfg(windows)]
+pub fn build_path(path: &str) -> PathBuf {
```

Comment:
Hm this seems very suspicious about how this would be called, how are MSYS paths leaking into the configuration here? Are the makefiles encoding MSYS paths?

Also, this can probably be done in Rust like [`sanitize_sh`](https://github.com/rust-lang/rust/blob/70b8b9487a94c5b9b1401007b040c584d9349582/src/bootstrap/build/dist.rs#L273-L290) rather than forking out to `cygpath`, right?